src/sentry/sentry_metrics/querying/metadata/tags.py

```diff
@@ -0,0 +1,60 @@
+from collections.abc import Sequence
+from dataclasses import dataclass
+
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+from sentry.snuba.metrics_layer.query import fetch_metric_tag_keys, fetch_metric_tag_values
+
+
+@dataclass
+class TagValue:
+    key: str
+    value: str
+
+    def __hash__(self):
+        return hash((self.key, self.value))
```

Comment:
Making the dataclass frozen would implement an hash method for you, I would generally implement your own hashing function if the object is mutable or your require equality to be defined on a subset of the fields:
```suggestion
@dataclass(frozen=True)
class TagValue:
    key: str
    value: str
```

Reply:
We should get rid of this type sooner than later, but at the moment all we need is for this type to be hashable to enable set logic. Adding the frozen keyword adds a level of meaning that would not be intended in my opinion.